standby/standby.go

```diff
@@ -2,29 +2,69 @@ package standby
 
 import (
 	"context"
+	"fmt"
 	"log/slog"
 
 	"github.com/EvergenEnergy/remote-standby/config"
+	mqtt "github.com/eclipse/paho.mqtt.golang"
 )
 
-type Worker struct {
-	logger *slog.Logger
-	cfg    config.Config
+type Service struct {
+	logger     *slog.Logger
+	cfg        config.Config
+	MQTTClient mqtt.Client
 }
 
-func Init(logger *slog.Logger, cfg config.Config) *Worker {
-	logger.Info("initing worker")
-	return &Worker{
-		logger: logger,
-		cfg:    cfg,
+func NewService(logger *slog.Logger, cfg config.Config) *Service {
+	return &Service{
+		logger:     logger,
+		cfg:        cfg,
+		MQTTClient: initMQTTClient(cfg),
 	}
 }
 
-func (w *Worker) Start(_ context.Context) {
-	w.logger.Info("started worker")
+var onConnect mqtt.OnConnectHandler = func(client mqtt.Client) {
+	fmt.Println("Connected")
 }
 
-func (w *Worker) Stop() error {
-	w.logger.Info("stopping worker")
+var onConnectionLost mqtt.ConnectionLostHandler = func(client mqtt.Client, err error) {
+	fmt.Printf("Connect lost: %v", err)
+}
+
+func initMQTTClient(cfg config.Config) mqtt.Client {
+	brokerURL := cfg.MQTT.BrokerURL
+
+	mqttOpts := mqtt.NewClientOptions()
+	mqttOpts.AddBroker(brokerURL)
+	mqttOpts.SetClientID("remote-standby-client")
+	mqttOpts.SetCleanSession(true)
+	mqttOpts.SetAutoReconnect(true)
+	mqttOpts.SetOrderMatters(true)
+	mqttOpts.SetOnConnectHandler(onConnect)
+	mqttOpts.SetConnectionLostHandler(onConnectionLost)
+
+	return mqtt.NewClient(mqttOpts)
+}
+
+func (s *Service) subscribeToTopic(topic string) {
+	token := s.MQTTClient.Subscribe(topic, 1,
+		func(client mqtt.Client, msg mqtt.Message) {
+			s.logger.Debug(fmt.Sprintf("Received message: %s from topic: %s", msg.Payload(), msg.Topic()))
+		})
+	token.Wait()
+	s.logger.Debug("Subscribed to topic " + topic)
+}
+
+func (s *Service) RunMQTT(ctx context.Context) error {
+	if token := s.MQTTClient.Connect(); token.Wait() && token.Error() != nil {
+		return token.Error()
+	}
+
+	s.subscribeToTopic(s.cfg.MQTT.StandbyTopic)
+
 	return nil
 }
+
+func (s *Service) StopMQTT() {
+	s.MQTTClient.Disconnect(uint(1000))
```

Comment:
cannot this fail? or return an error? if so, we should probably propagate this.

Reply:
The `Disconnect` method doesn't return anything, so I guess there's no way to know if it fails